internal/provider/models/team.go

```diff
@@ -64,3 +80,55 @@ func (data *Team) ReadFromResponse(ctx context.Context, team *iam.Team) diag.Dia
 
 	return nil
 }
+
+func (data *TeamResource) ReadFromResponse(ctx context.Context, team *iam.Team, memberIds *[]string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	data.Id = types.StringValue(team.Id)
+	data.Name = types.StringValue(team.Name)
+	if team.Description != nil && *team.Description != "" {
+		data.Description = types.StringValue(*team.Description)
+	} else {
+		data.Description = types.StringNull()
+	}
+	if memberIds != nil && len(*memberIds) > 0 {
```

Comment:
prob dont need this if statement since the util function covers it

Reply:
the issue i ran into was that the util function returns `types.SetValue(types.StringType, []attr.Value{})` for nil values

but what i need is a null set, which is `types.SetNull(types.StringType)`

Reply:
i'll keep testing with and without the guard but i ran into issues when i didn't do it this way

Reply:
makes sense
can leave it as is if it errors